### PR TITLE
feat(storybook): cypress setup for the project

### PR DESCRIPTION
**Pull request by sanketshevkar**
Signed-off-by: User <shevkar.sanket@gmail.com>

### Changes

- Added cypress as a devDependency and wrote a dummy test within the cypress/intergrations folder.
- You need to install cypress on your local machine so as to run tests. Instructions for installing cypress on local machine: https://docs.cypress.io/guides/getting-started/installing-cypress.html#System-requirements
- To run the test you need two terminal windows, on the first one you need to run `npm run storybook` and once the storybook loads, on the second terminal within packages/storybook run `npm run cypress:open`. Cypress app would start and select the markdownEditor.spec.js file and then the test should start.

### Flags

- You might have to run `lerna clean && lerna bootstrap && lerna run build` as a new devDependency is added.

### Screenshots or Video

https://www.loom.com/share/72e05a6e0c014ba6940fb36b757580a8

### Related Issues
- Issue #14 

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `master` from `fork:branchname`
- [ ] Manual accessibility test performed
    - [ ] Keyboard-only access, including forms
    - [ ] Contrast at least WCAG Level A
    - [ ] Appropriate labels, alt text, and instructions

### CODE REVIEW

**Comment by irmerk** [comment]
@mttrbrts and I are looking into this from https://github.com/accordproject/web-components/issues/279

**Review by Cronus1007 — CHANGES_REQUESTED** [review]

File: packages/storybook/package.json [hunk 1]
```diff
@@ -58,7 +58,8 @@
     "storybook": "start-storybook -p 9009 -s public",
     "build-storybook": "NODE_OPTIONS='--max-old-space-size=4096' build-storybook -s public --quiet",
     "build-storybook:watch": "NODE_OPTIONS='--max-old-space-size=4096' build-storybook -w -s public --quiet",
-    "test": "NODE_ENV=test jest"
+    "test": "NODE_ENV=test jest",
+    "cypress:open": "cypress open"
```
Comment (Cronus1007) [inline packages/storybook/package.json:62]: @sanketshevkar The script name should rather reflect the work associated with the script so would be better to change the name to `test:cypress` or `test:e2e`.

**Comment by sanketshevkar** [comment]
> @mttrbrts and I are looking into this from #279

Okay cool. Should I close this pr?

**Comment by mttrbrts** [comment]
> > @mttrbrts and I are looking into this from #279
> 
> Okay cool. Should I close this pr?

No, I've added another commit which completes the basic setup. @rkotangoor @irmerk  can you review, please?

**Comment by mttrbrts** [comment]
In theory, Cypress should report the test results as comments in PRs, and the history will be publicly available at https://dashboard.cypress.io/projects/vrh5u7

**Comment by Cronus1007** [comment]
@mttrbrts I have a question that the tests should be controlled via the main directory rather than the storybook package.

**Comment by mttrbrts** [comment]
> @mttrbrts I have a question that the tests should be controlled via the main directory rather than the storybook package. Whats your opinion regarding this?

That's a good question. For these E2E tests, we want to test the components in context (i.e. in an app), and so storybook is a convenient tool for that. I agree that we should be able to run the tests from a `package.json` script at the root-level too (I'll add it to this PR).

Unit tests for example are better suited to being situated local to the module / package that they are testing.

**Comment by Cronus1007** [comment]
@mttrbrts For unit tests you can have a link at this comment https://github.com/accordproject/web-components/issues/14#issuecomment-796096658

**Comment by rkotangoor** [comment]
@sanketshevkar : could you add instructions to storybook Readme on how to run the cypress tests? Pretty much what you have included in this PR description?

**Comment by sanketshevkar** [comment]
> @sanketshevkar : could you add instructions to storybook Readme on how to run the cypress tests? Pretty much what you have included in this PR description? 

Sure would do that 👍.

**Comment by sanketshevkar** [comment]
@mttrbrts @rkotangoor I've updated the readme file.

**Comment by DianaLease** [comment]
> This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/vrh5u7/settings?utm_source=github)

This is awesome. @mttrbrts what kicks off the test run? Whenever a new PR is opened?

**Comment by mttrbrts** [comment]
> > This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/vrh5u7/settings?utm_source=github)
> 
> This is awesome. @mttrbrts what kicks off the test run? Whenever a new PR is opened?

Thanks, yes, here's the config from cypress.io 
image

**Comment by Cronus1007** [comment]
@mttrbrts Shall we update this to Travis CI/CD pipeline plus `codecov` report to the repository whenever a new PR is opened.

**Review by irmerk — CHANGES_REQUESTED** [review]
Needs investigation by @accordproject/maintainers-ui or @mttrbrts.

**Comment by mttrbrts** [comment]
> @mttrbrts Shall we update this to Travis CI/CD pipeline plus `codecov` report to the repository whenever a new PR is opened.

This is a separate issue. Let's focus on getting this change in first.

**Review by mttrbrts — CHANGES_REQUESTED** [review]
@sanketshevkar can you fix `package-lock.json`, please? Otherwise, we're ready to merge this.

**Comment by sanketshevkar** [comment]
> @sanketshevkar can you fix `package-lock.json`, please? Otherwise, we're ready to merge this.
I'm not able to find where the merge conflict is exactly located. Can someone help me out?

**Comment by Cronus1007** [comment]
@sanketshevkar First of all update your branch. This will remove the `merge-conflicts` .

**Comment by sanketshevkar** [comment]
@mttrbrts why is it expecting a dco of mine from some other id. I've made all my dco sign off's using shevkar.sanket@gmail.com.

**Comment by mttrbrts** [comment]
> @mttrbrts why is it expecting a dco of mine from some other id. I've made all my dco sign off's using [shevkar.sanket@gmail.com](mailto:shevkar.sanket@gmail.com).

Apologies, I've been trying (and failing!) to fix this properly. I've now manually reviewed and approved the commits.

**Review by mttrbrts — APPROVED** [review]

**Comment by sanketshevkar** [comment]
@mttrbrts I've updated my branch and created new package-lock.json file. Should I commit that?

**Comment by sanketshevkar** [comment]
> > @mttrbrts why is it expecting a dco of mine from some other id. I've made all my dco sign off's using [shevkar.sanket@gmail.com](mailto:shevkar.sanket@gmail.com).
> 
> 
> 
> Apologies, I've been trying (and failing!) to fix this properly. I've now manually reviewed and approved the commits. 

No, problem. Thanks a lot for helping out.

**Comment by irmerk** [comment]
So, as of right now, we manually run this locally and it will produce a test result in Cypress? Next steps will be:
- Set up automatic runs in the Travis (soon to be GitHub Actions) scripts to potentially run in each PR? 
- Continue to iterate on the Cypress test suite?

**Comment by mttrbrts** [comment]
> So, as of right now, we manually run this locally and it will produce a test result in Cypress? Next steps will be:
> 
> * Set up automatic runs in the Travis (soon to be GitHub Actions) scripts to potentially run in each PR?
> * Continue to iterate on the Cypress test suite?

Yes, that's right. Although the PR integration is already setup (without using Travis or GH Actions). See https://github.com/accordproject/web-components/pull/282#issuecomment-797491849

**Review by irmerk — APPROVED** [review]
👏🏻

**Pull request merged**